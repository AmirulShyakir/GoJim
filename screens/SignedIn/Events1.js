import { db } from "../../firebase";
import {
  collection,
  doc,
  getDoc,
  getDocs,
  query,
  where,
} from "firebase/firestore";
import MainContainer from "../../components/containers/MainContainer";
import SignedInContainer from "../../components/containers/SignedInContainer";
//texts
import LargeText from "../../components/Texts/LargeText";
import RegularButton from "../../components/Buttons/RegularButton";
import { FlatList } from "react-native";
import EventCard from "../../components/containers/EventCard";
import EventTypeCard from "../../components/containers/EventTypeCard";
import EventContainer from "../../components/containers/EventContainer";
import { useState, useEffect } from "react";

const Events1 = ({ navigation }) => {
  const [selected, setSelected] = useState("");

  useEffect(() => {
    console.log(selected + "joe");
  }, [selected]);

  function pressEvent(eventType) {
    setSelected(eventType);
    console.log(selected);
    navigation.navigate("EventsScreen", { eventType: selected });
  }

  return (
    <EventContainer>
      <EventTypeCard
        onPress={() => {
          pressEvent("Friendly Sports");
        }}
        image={{
          uri: "https://images.unsplash.com/photo-1616453239004-3ab56827fb76?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8MTh8fHRlbm5pcyUyMHBsYXllcnxlbnwwfHwwfHw%3D&w=1000&q=80",
        }}
      >
        Friendly Sports
      </EventTypeCard>

      <EventTypeCard
        onPress={() => {
          pressEvent("Networking Sessions");
        }}
        image={{
          uri: "https://media.istockphoto.com/photos/brilliant-ideas-in-the-making-picture-id540526464?k=20&m=540526464&s=612x612&w=0&h=Gz0d2rwaT7Z9OxAPxrHRcddVVblvQt77C-pqvX4pcAw=",
        }}
      >
        Networking Sessions
      </EventTypeCard>
      <EventTypeCard
        onPress={() => {
          pressEvent("Recreational Training");
        }}
        image={{
          uri: "https://vivace.smu.edu.sg/sites/vivace.smu.edu.sg/files/styles/gallery/public/2021-08/euru%204.JPG?itok=fRtmCk5Y",
        }}
      >
        Recreational Training
      </EventTypeCard>
      <EventTypeCard
        onPress={() => {
          pressEvent("Study Sessions");
        }}
        image={{
          uri: "https://images.unsplash.com/photo-1604882735652-699f1c867bba?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=387&q=80",
        }}
      >
        Study Sessions
      </EventTypeCard>
    </EventContainer>
  );
};

export default Events1;
