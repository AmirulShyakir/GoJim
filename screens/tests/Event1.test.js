import React from "react";
import renderer from 'react-test-renderer';
import { render, fireEvent } from "@testing-library/react-native";
import Events1 from "../SignedIn/Events1";

test("Event1 snapshot test", () => {
  const tree = renderer.create(<Events1 />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("it renders text correctly", () => {
  const { getByText, queryAllByText }= render(<Events1 />);
  expect (queryAllByText("Friendly Sports").length).not.toBe(0);
  expect (queryAllByText("Networking Sessions").length).not.toBe(0);
  expect (queryAllByText("Recreational Training").length).not.toBe(0);
  expect (queryAllByText("Study Sessions").length).not.toBe(0);
})
