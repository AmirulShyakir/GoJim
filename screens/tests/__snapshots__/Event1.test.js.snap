// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event1 snapshot test 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#413636",
      "flex": 1,
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "center",
      "padding": 5,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#1c1515",
        "borderRadius": 10,
        "height": "48%",
        "margin": 5,
        "opacity": 1,
        "width": "46%",
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Array [
          Object {
            "borderRadius": 10,
            "height": "100%",
            "opacity": 50,
            "width": "100%",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "https://images.unsplash.com/photo-1616453239004-3ab56827fb76?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxzZWFyY2h8MTh8fHRlbm5pcyUyMHBsYXllcnxlbnwwfHwwfHw%3D&w=1000&q=80",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            Array [
              Object {
                "borderRadius": 10,
                "height": "100%",
                "opacity": 50,
                "width": "100%",
              },
            ],
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 25,
              "paddingBottom": 10,
              "paddingTop": 10,
              "textAlign": "left",
            },
            Object {
              "fontWeight": "bold",
              "padding": 10,
            },
          ]
        }
      >
        Friendly Sports
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#1c1515",
        "borderRadius": 10,
        "height": "48%",
        "margin": 5,
        "opacity": 1,
        "width": "46%",
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Array [
          Object {
            "borderRadius": 10,
            "height": "100%",
            "opacity": 50,
            "width": "100%",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "https://media.istockphoto.com/photos/brilliant-ideas-in-the-making-picture-id540526464?k=20&m=540526464&s=612x612&w=0&h=Gz0d2rwaT7Z9OxAPxrHRcddVVblvQt77C-pqvX4pcAw=",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            Array [
              Object {
                "borderRadius": 10,
                "height": "100%",
                "opacity": 50,
                "width": "100%",
              },
            ],
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 25,
              "paddingBottom": 10,
              "paddingTop": 10,
              "textAlign": "left",
            },
            Object {
              "fontWeight": "bold",
              "padding": 10,
            },
          ]
        }
      >
        Networking Sessions
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#1c1515",
        "borderRadius": 10,
        "height": "48%",
        "margin": 5,
        "opacity": 1,
        "width": "46%",
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Array [
          Object {
            "borderRadius": 10,
            "height": "100%",
            "opacity": 50,
            "width": "100%",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "https://vivace.smu.edu.sg/sites/vivace.smu.edu.sg/files/styles/gallery/public/2021-08/euru%204.JPG?itok=fRtmCk5Y",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            Array [
              Object {
                "borderRadius": 10,
                "height": "100%",
                "opacity": 50,
                "width": "100%",
              },
            ],
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 25,
              "paddingBottom": 10,
              "paddingTop": 10,
              "textAlign": "left",
            },
            Object {
              "fontWeight": "bold",
              "padding": 10,
            },
          ]
        }
      >
        Recreational Training
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#1c1515",
        "borderRadius": 10,
        "height": "48%",
        "margin": 5,
        "opacity": 1,
        "width": "46%",
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Array [
          Object {
            "borderRadius": 10,
            "height": "100%",
            "opacity": 50,
            "width": "100%",
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "https://images.unsplash.com/photo-1604882735652-699f1c867bba?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=387&q=80",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            Array [
              Object {
                "borderRadius": 10,
                "height": "100%",
                "opacity": 50,
                "width": "100%",
              },
            ],
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontSize": 25,
              "paddingBottom": 10,
              "paddingTop": 10,
              "textAlign": "left",
            },
            Object {
              "fontWeight": "bold",
              "padding": 10,
            },
          ]
        }
      >
        Study Sessions
      </Text>
    </View>
  </View>
</View>
`;
